src/app/common/filter-text/filter-text.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class FilterTextService {
+  constructor() {}
+
+  filter(data: string, props: Array<string>, originalList: Array<any>) {
+    let filteredList = [];
+
+    if (data && props && originalList) {
+      data = data.toLowerCase();
+
+      originalList.filter(item => {
+        let match = false;
+
+        for (const prop of props) {
+
+          console.log('prop: ' + JSON.stringify(prop));
+
+          if (item[prop].toString().toLowerCase().indexOf(data) > -1) {
+            console.log('match found: ' + JSON.stringify(item[prop]));
+            match = true;
+
+            filteredList.push(item);
+
+            console.log('match: ' + JSON.stringify(match));
+            console.log('item: ' + JSON.stringify(item));
+            break;
+          }
+
+          console.log('filteredList: ' + JSON.stringify(filteredList));
```

Comment:
nuther one, also for really long lists this will cause a lot of text to go to the dev console, better to not use JSON.stringify() and let the browser render the array, like:

```
console.log('filteredList: ', filteredList);
```